### DISCUSSION TITLE
Implement or declare OnDocuments and OnError

### DISCUSSION THREAD

**Discussion by julesledev** [SchneiderInfosystems/FB4D#123] [Q&A] [answered]
I would like to implement OnDocuments, OnError in order to get documents from firestore but i don't know how to proceed

![image](https://user-images.githubusercontent.com/71714827/146584763-afd5e67b-68af-4271-ae1f-8efee8a28636.png)

And I would like to retrieve all the collection data

Thanks

**Comment by SchneiderInfosystems** [comment, marked as answer]
Have you checked the Intro Demo Application first?
In Line 1115 of https://github.com/SchneiderInfosystems/FB4D/blob/master/Samples/Intro/FB4D.DemoFmx.pas we have a such call.

This is is the shortest version of such `OnDocuments` callback function:

```
procedure TfmxFirebaseDemo.OnFirestoreGet(const Info: string;
  Docs: IFirestoreDocuments);
var
  Doc: IFirestoreDocument;
begin
  if assigned(Docs) and (Docs.Count > 0) then
    for Doc in Docs do
      ShowDocument(Doc);
end;
```
